mongo/client_examples_test.go

```diff
@@ -314,3 +314,50 @@ func ExampleConnect_aWS() {
 	}
 	_ = ecClient
 }
+
+func ExampleConnect_versionedAPI() {
```

Comment:
I wonder if this is the best place for this documentation. I could put it directly below the `mongo` package-level documentation of client side encryption [here](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/doc.go#L144) instead.

Reply:
This seems ok to me, but I will defer judgement to the Divjot/Isabella.

I do like that these examples are compiled, instead of having them in doc comments. Example code in doc comments adds risk that we write non-compilable examples.